src/Umbraco.Web.Common/Extensions/LinkGeneratorExtensions.cs

```diff
@@ -12,33 +12,36 @@ namespace Umbraco.Extensions;
 
 public static class LinkGeneratorExtensions
 {
+    /// <summary>
+    /// Gets the Umbraco backoffice URL (if Umbraco is installed).
+    /// </summary>
+    /// <param name="linkGenerator">The link generator.</param>
+    /// <returns>
+    /// The Umbraco backoffice URL.
+    /// </returns>
+    public static string? GetUmbracoBackOfficeUrl(this LinkGenerator linkGenerator)
+        => linkGenerator.GetPathByAction("Default", "BackOffice", new { area = Constants.Web.Mvc.BackOfficeArea });
+
+    /// <summary>
+    /// Gets the Umbraco backoffice URL (if Umbraco is installed) or application virtual path (in most cases just <c>"/"</c>).
+    /// </summary>
+    /// <param name="linkGenerator">The link generator.</param>
+    /// <param name="hostingEnvironment">The hosting environment.</param>
+    /// <returns>
+    /// The Umbraco backoffice URL.
+    /// </returns>
+    public static string GetUmbracoBackOfficeUrl(this LinkGenerator linkGenerator, IHostingEnvironment hostingEnvironment)
+         => GetUmbracoBackOfficeUrl(linkGenerator) ?? hostingEnvironment.ApplicationVirtualPath;
+
     /// <summary>
     ///     Return the back office url if the back office is installed
     /// </summary>
+    /// <remarks>
+    /// This method contained a bug that would result in always returning "/".
+    /// </remarks>
+    [Obsolete("Use the GetUmbracoBackOfficeUrl extension method instead. This method will be removed in Umbraco 13.")]
     public static string? GetBackOfficeUrl(this LinkGenerator linkGenerator, IHostingEnvironment hostingEnvironment)
-    {
-        Type? backOfficeControllerType;
-        try
-        {
-            backOfficeControllerType = Assembly.Load("Umbraco.Web.BackOffice")
-                .GetType("Umbraco.Web.BackOffice.Controllers.BackOfficeController");
```

Comment:
This will always be `null`, because this assembly doesn't have the `Umbraco.Web.BackOffice.Controllers.BackOfficeController` type, as the namespace changed and it's now `Umbraco.Cms.Web.BackOffice.Controllers.BackOfficeController`:

https://github.com/umbraco/Umbraco-CMS/blob/4d23e8a6bd1c5e8637089a7c5d49cdffbf7495d6/src/Umbraco.Web.BackOffice/Controllers/BackOfficeController.cs#L37-L43